lib/screens/home/widget/overlay_content.dart

```diff
@@ -1,70 +1,137 @@
+                    height: 25,
+                    width: 25,
                   ),
-                ),
-                Spaces.h15,
-                SizedBox(
-                  height: 60,
-                  width: double.infinity,
-                  child: GestureDetector(
+                  Spaces.w20,
+                  Expanded(
+                    child: TextFormField(
+                      style: const TextStyle(color: CustomColors.textColor),
+                      controller: _textFieldController,
+                      decoration: const InputDecoration(
+                        hintText: 'Enter your text',
+                        hintStyle: TextStyle(
+                          color: Color.fromARGB(255, 116, 115, 115),
+                        ),
+                        border: InputBorder.none,
+                      ),
+                    ),
+                  ),
+                  Spaces.h15,
+                  GestureDetector(
                     onTap: () {
-                      DataWriter dataWriter = DataWriter(
-                          hasCompleted: false,
-                          taskContent: _textFieldController.text);
-                      dataWriter.create();
-                      OverlayManager.removeOverlay();
+                      widget.isEditing ? taskUpdation() : newTaskCreation();
                     },
-                    child: Container(
-                      padding: const EdgeInsets.all(10),
-                      alignment: Alignment.centerRight,
-                      height: 30,
-                      width: 70,
-                      child: const Text(
-                        'Done',
-                        style: TextStyle(
-                          color: CustomColors.lightGrey,
-                          fontSize: 18,
-                          fontWeight: FontWeight.bold,
+                    child: SizedBox(
+                      height: 60,
+                      width: 90,
+                      child: Container(
+                        padding: const EdgeInsets.all(10),
+                        alignment: Alignment.centerRight,
+                        height: 30,
+                        width: 70,
+                        child: Text(
+                          'Done',
+                          style: TextStyle(
+                            color: isTextFieldEmpty
+                                ? CustomColors.lightGrey
+                                : CustomColors.yellow,
+                            fontSize: 18,
+                            fontWeight: FontWeight.bold,
+                          ),
                         ),
                       ),
                     ),
                   ),
-                ),
-              ],
+                ],
+              ),
             ),
           ),
         ),
       ),
     );
   }
+
+  void newTaskCreation() {
+    DataWriter dataWriter = DataWriter(
+      hasCompleted: false,
```

Comment:
these arent the properties of data writer, these should be the property of a task model, please make a seperate layer for this